bifromq-mqtt/bifromq-mqtt-broker-client/src/main/java/com/baidu/bifromq/mqtt/inbox/MqttBrokerClient.java

```diff
@@ -127,7 +127,32 @@ public CompletableFuture<DeliveryReply> deliver(DeliveryRequest request) {
                     .setReqId(reqId)
                     .setRequest(request)
                     .build())
-                .thenApply(WriteReply::getReply);
+                .thenApply(WriteReply::getReply)
+                .handle((reply, ex) -> {
+                    if (ex != null) {
+                        if (ex.getCause() instanceof ServerUnreachableException
+                                || ex.getCause() instanceof ServerNotFoundException) {
+                            DeliveryReply.Builder replyBuilder = DeliveryReply.newBuilder();
+                            Map<String, DeliveryResults> resultsMap = new HashMap<>();
+                            for (Map.Entry<String, DeliveryPackage> entry : request.getPackageMap().entrySet()) {
+                                String tenantId = entry.getKey();
+                                List<DeliveryResult> deliveryResults = new ArrayList<>();
+                                for (DeliveryPack pack: entry.getValue().getPackList()) {
+                                    for (MatchInfo matchInfo : pack.getMatchInfoList()) {
+                                        deliveryResults.add(DeliveryResult.newBuilder()
```

Comment:
`matchInfos` need to be deduplicated before adding to `deliveryResults`, since same matchInfo may appears multiple times in the `DeliveryRequest`